src/main/java/graphql/schema/transform/VisibleFieldPredicate.java

```diff
@@ -0,0 +1,23 @@
+package graphql.schema.transform;
+
+
+import graphql.PublicSpi;
+import graphql.schema.GraphQLFieldDefinition;
+
+/**
+ * Predicate used during a {@link FieldVisibilitySchemaTransformation} to test whether a field should be visible.
+ */
+@PublicSpi
+@FunctionalInterface
+public interface VisibleFieldPredicate {
+
+    /**
+     * Test whether a field should be visible. Provided as a more descriptive "test" method that describes exactly
+     * what a positive result of "test" should mean.
+     *
+     * @param definition field definition
+     * @param environment the context of the field
+     * @return true if visible
+     */
+    boolean isVisible(GraphQLFieldDefinition definition, VisibleFieldPredicateEnvironment environment);
```

Comment:
Sorry Dan - maybe we should have been more explicit.

There should be only 1 VisibleFieldPredicateEnvironment parameter that contains 2 properties

parentElement
fieldDefinition

Can we use this name please.  We have AST nodes in graphql-java and we also have the runtime graphql schema elements which are also nodes of a sort.

We tend to use node as names for AST things.   There is also AST definitions.

So can we name them fieldDefinition and parentElement please

Reply:
Ah ok gotcha. Will update.

Reply:
Done.